sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -338,6 +342,9 @@ export class BatchingReceiverLite {
     const loggingPrefix = `[${receiver.connection.id}|r:${receiver.name}]`;
 
     let totalWaitTimer: NodeJS.Timer | undefined;
+    let drainPromiseResolver: (value: unknown) => void;
+    const drainPromise = new Promise((resolve) => (drainPromiseResolver = resolve));
+    const drainTimeoutAborter = new AbortController();
```

Comment:
I feel that this PR may also be changed to use `setTimeout()/clearTimeout()` pair.

Reply:
I was able to use setTimeout and clearTimeout instead of relying on abort error.  The complexity of new code looks similarly though because of the need to resolve outside in the close handler.